virt_who/provision.py

```diff
@@ -2058,10 +2058,17 @@ def vcenter_guest_uuid(self, cert, ssh_vcenter, guest_name):
         ret, output = self.runcmd(cmd, ssh_vcenter)
         if ret == 0:
             for line in output.splitlines():
+                if re.match(r"^Version.*:", line):
+                    version = line.split(':')[1].strip()
+                    logger.info("Succeeded to get the VMware version: {0}".format(version))
                 if re.match(r"^Uuid.*:", line):
-                    uuid = line.split(':')[1].strip()
-                    logger.info("Succeeded to get vcenter guest uuid: {0}".format(uuid))
-                    return uuid
+                    s = line.split(':')[1].strip()
+            if version > "vmx-13":
+                uuid = s[6:8] + s[4:6] + s[2:4] + s[0:2] + "-" + s[11:13] + s[9:11] + "-" + s[16:18] + s[14:16] + "-" + s[19:]
+            else:
+                uuid = s
```

Comment:
There is a problem:  if `version` or  `uuid` are not exist in the output,  it will give you an exception: version or uuid is not defined, consider the following update:)
```        version=''
            uuid=''
            for line in output.splitlines():
                if re.match(r"^Version.*:", line):
                    version = line.split(':')[1].strip()
                if re.match(r"^Uuid.*:", line):
                    uuid = line.split(':')[1].strip()
            if uuid:
                if version > "vmx-13":
                    return uuid[6:8] + uuid[4:6] + uuid[2:4] + uuid[0:2] + "-" + uuid[11:13] + uuid[9:11] + "-" + uuid[16:18] + uuid[14:16] + "-" + uuid[19:]
                else:
                    return uuid
            else:
                raise FailException("Failed to get vcenter guest uuid")
```

Reply:
Done it, thanks!

Reply:
if there is no Uuid found by `if re.match(r"^Uuid.*:", line):`, what happen?

Reply:
added ``` if uuid and version:``` to make make sure getting the two values.

Reply:
The changes is good, but I still have a little concern: if uuid exist, but version is none or version < 13, we should return uuid.

Reply:
I removed checking the version value , please review again.